nautobot/extras/tests/test_views.py

```diff
@@ -1975,10 +1977,74 @@ def test_required_relationships(self):
         1. Try creating an object when no required target object exists
         2. Try creating an object without specifying required target object(s)
         3. Try creating an object when all required data is present
+        4. Test bulk edit
         """
-        # Parameterized test:
+
+        # Parameterized tests (for creating and updating single objects):
         self.required_relationships_test(interact_with="ui")
 
+        # 4. Bulk create/edit tests:
+
+        vlans = VLANFactory.create_batch(6)
+
+        # Try deleting all devices and then editing the 6 VLANs (fails):
+        Device.objects.all().delete()
+        response = self.client.post(
+            reverse("ipam:vlan_bulk_edit"), data={"pk": [str(vlan.id) for vlan in vlans], "_apply": [""]}
+        )
+        self.assertContains(response, "VLANs require at least one device, but no devices exist yet.")
+
+        # Create test device for association
+        device_for_association = test_views.create_test_device("VLAN Required Device")
+
+        # Try editing all 6 VLANs without adding the required device(fails):
+        response = self.client.post(
+            reverse("ipam:vlan_bulk_edit"), data={"pk": [str(vlan.id) for vlan in vlans], "_apply": [""]}
+        )
+        self.assertContains(
+            response,
+            "6 VLANs require a device for the required relationship &quot;VLANs require at least one Device&quot;",
+        )
+
+        # Try editing 3 VLANs without adding the required device(fails):
+        response = self.client.post(
+            reverse("ipam:vlan_bulk_edit"), data={"pk": [str(vlan.id) for vlan in vlans[:3]], "_apply": [""]}
+        )
+        self.assertContains(
+            response, 'These VLANs require a device for the required relationship "VLANs require at least one Device":'
```

Comment:
Why `&quot;` in the previous response, but `"` in this response?

Reply:
I'm not sure why it renders the HTML differently in each case. Will look into it!

Reply:
I re-wrote some of the logic and now this is a non-issue!